### PR TITLE
Delete a Single Note

### PR DESCRIPTION
**Pull request by uhalder**
This PR has the following changes
1) In the view notes screen, a delete icon is added for each individual note. Clicking the delete icon will show a confirmation dialog and delete the note.
2) In the edit notes screen, a delete note button is added for each individual note. Clicking the button will again show a confirmation dialog and delete the note.

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -71,6 +71,11 @@
     <string name="edit_interview_header">Update your scheduled interview details.</string>
     <string name="add_note_question">Add Question</string>
     <string name="add_note">Add Note</string>
+    <string name="appbar_header_add_notes">Add Notes</string>
+    <string name="appbar_header_edit_notes">Edit Notes</string>
```
Comment (Audienix) [inline app/src/main/res/values/strings.xml]: Can we delete these two ? These two are unused and redundant as the strings are defined already in this file. Refer 
`appbar_title_add_notes`
`appbar_title_edit_notes   `

File: app/src/main/kotlin/com/twain/interprep/domain/usecase/note/DeleteNoteUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+import com.twain.interprep.data.model.Note
+import com.twain.interprep.domain.repository.NoteRepository
+
+class DeleteNoteUseCase(private val noteRepository: NoteRepository) {
+    suspend operator fun invoke(
+        note: Note
+    ) = noteRepository.deleteNote(note)
+
+}
```
Comment (Audienix) [inline app/src/main/kotlin/com/twain/interprep/domain/usecase/note/DeleteNoteUseCase.kt]: Typically 1 empty line is given at the end of file.

File: app/src/main/kotlin/com/twain/interprep/presentation/ui/components/note/AddNoteCard.kt [hunk 3]
```diff
@@ -63,27 +69,48 @@ fun AddNoteCard(
...
+                        backgroundColor = BackgroundPalePurple,
+                        textColor = Color.Black,
+                        text = stringResource(id = R.string.add_note_question),
+                        iconColor = BackgroundDarkPurple,
+                        leadingIcon = R.drawable.outline_add_circle,
+                        onClick = { addQuestion() },
+                        borderColor = BackgroundDarkPurple,
+                        textStyle = MaterialTheme.typography.titleMedium,
+                        contentPadding = PaddingValues(
+                            horizontal = dimensionResource(id = R.dimen.dimension_16dp)
```
Comment (Audienix) [inline app/src/main/kotlin/com/twain/interprep/presentation/ui/components/note/AddNoteCard.kt]: nit: This if-else could be optimized avoiding repetition of lines ?

File: app/src/main/kotlin/com/twain/interprep/presentation/ui/components/note/DeleteNoteDialog.kt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+            onPositiveButtonClick = {
+                showDeleteDialog.value = false
+                deleteNote()
+            }, // "OK" is clicked
+            onNegativeButtonClick = {
+                showDeleteDialog.value = false
+            } // "CANCEL" is clicked
+        )
+    }
+}
```
Comment (Audienix) [inline app/src/main/kotlin/com/twain/interprep/presentation/ui/components/note/DeleteNoteDialog.kt]: Empty new line is missing.

**Review by Audienix — APPROVED** [review]
Some minor feedback, rest all looking good.

**Comment by Audienix** [comment]
@uhalder - Also Figma is updated for both these cases, you can do these in next PR.

1. Instead of showing delete and edit button in ViewNoteCard in a Column, show a menu option in the card, on click of that menu option, show a  popup menu which shows "Edit Note" and "Delete Note" options. [Figma Link](https://www.figma.com/file/SptJLDZmGfgT1NJruX88PK/InterPrep?type=design&node-id=171-4588&mode=design&t=nGQIKkV2n5iRztX6-0)
image

2. In both the cases of View Notes & Edit Notes, after deleting all notes, show empty state. [Figma Link](https://www.figma.com/file/SptJLDZmGfgT1NJruX88PK/InterPrep?type=design&node-id=971-4519&mode=design&t=nGQIKkV2n5iRztX6-0)
image

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (uhalder) [reply app/src/main/res/values/strings.xml]: This is done

**Comment by Audienix** [comment]
@uhalder Thanks for the fixes. Request you to squash & merge the Pull Request from next time. It keeps development branch history clean and easy to follow.

**Pull request merged**